src/wakefs/fuse.cpp

```diff
@@ -227,35 +222,41 @@ bool run_in_fuse(fuse_args &args, int &status, std::string &result_json) {
   (void)close(STDOUT_FILENO);
   (void)close(STDERR_FILENO);
 
-  int pidfd = syscall(SYS_pidfd_open, pid, 0);
-  if (pidfd < 0) {
-    std::cerr << "wakebox: Failed to open fd for pid" << std::endl;
-    exit(1);
-  }
+  bool has_timeout = args.command_timeout > 0;
+  pid_t timeout_pid = -1;
 
-  EPoll epoll;
-  epoll.add(pidfd, EPOLLIN);
+  // Launch timer process
+  if (has_timeout) {
+    timeout_pid = fork();
+    if (timeout_pid < 0) {
+      std::cerr << "wakebox: failed to fork timout process" << std::endl;
+      exit(1);
+    }
 
-  std::vector<epoll_event> events;
-  if (args.command_timeout <= 0) {
-    events = epoll.wait(nullptr, nullptr);
-  } else {
-    struct timespec timeout;
-    timeout.tv_sec = args.command_timeout;
-    timeout.tv_nsec = 0;
-    events = epoll.wait(&timeout, nullptr);
+    if (timeout_pid == 0) {
+      std::string sleep = "sleep " + std::to_string(args.command_timeout);
+      execve_wrapper({"/bin/sh", "-c", sleep}, {});
+    }
   }
 
-  if (events.empty()) {
-    kill(pid, SIGKILL);
-    std::cerr << "wakebox: Timed out waiting for process to complete" << std::endl;
-    exit(124);
-  }
+  pid_t wait_pid;
+  while ((wait_pid = wait(&status)) != -1) {
+    if (wait_pid == timeout_pid && WIFEXITED(status)) {
+      kill(payload_pid, SIGKILL);
+      std::cerr << "wakebox: Timed out waiting for process to complete" << std::endl;
+      exit(124);
```

Comment:
I think we still need to output the results so it doesn't suffice to exit here. Data still has to be collected or Nathan's use case is not covered.